osm2streets/src/transform/classify_intersections.rs

```diff
@@ -1,85 +1,229 @@
 use crate::osm::NodeID;
+use crate::types::Movement;
+use crate::Direction;
 use crate::IntersectionComplexity::*;
-use crate::{Direction, IntersectionComplexity, StreetNetwork};
+use crate::{
+    ConflictType, DrivingSide, IntersectionComplexity, RestrictionType, Road, StreetNetwork,
+};
+use std::cmp::{max, min};
+use std::collections::BTreeMap;
 
 /// Determines the initial complexity of all intersections. Intersections marked "Crossing" are
 /// considered "unclassified" and will be updated with a guess, others will be left unchanged.
 pub fn classify_intersections(streets: &mut StreetNetwork) {
-    let mut changes: Vec<(NodeID, IntersectionComplexity)> = Vec::new();
+    let mut changes: Vec<_> = Vec::new();
     for (id, inter) in &streets.intersections {
         if let Crossing = inter.complexity {
             changes.push((*id, guess_complexity(streets, id)));
         }
     }
 
-    for (id, complexity) in changes.into_iter() {
-        streets.intersections.get_mut(&id).unwrap().complexity = complexity;
+    for (id, (complexity, conflict_level, movements)) in changes {
+        let intersection = streets.intersections.get_mut(&id).unwrap();
+        intersection.complexity = complexity;
+        intersection.conflict_level = conflict_level;
+        intersection.movements = movements;
     }
 }
 
 /// Guesses the complexity of the intersection based on the connecting roads and their lanes.
 ///
 /// The existing complexity field is ignored, so be careful how you use the guessed value.
-fn guess_complexity(streets: &StreetNetwork, intersection_id: &NodeID) -> IntersectionComplexity {
-    let roads = streets.roads_per_intersection(*intersection_id);
+pub fn guess_complexity(
+    streets: &StreetNetwork,
+    intersection_id: &NodeID,
+) -> (IntersectionComplexity, ConflictType, Vec<Movement>) {
+    use ConflictType::*;
+    let roads: Vec<_> = streets
+        .roads_per_intersection(*intersection_id)
```

Comment:
I was going to propose this is another cleanup PR... what do you think about `roads_per_intersection` returning `Vec<&Road>`? Many places need to do something like this. Those that just need road IDs can use the `roads` field that's now just stored in the struct

Reply:
I'm for it!